### PR TITLE
detect/transform: from_base64 transform 

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #10972 

This PR provides the `from_base64` transform used to match on content decoded with base64 using the mode (default rfc4648).

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [6487](https://redmine.openinfosecfoundation.org/issues/6487)

Describe changes:
- New transform: `from_base64`
- Documentation
- Keyword parser in rust.

Updates:
- Fix bug that caused validation assert.

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1670

### CODE REVIEW

**Comment by jufajardini** [comment]
Wanted to re-run failed (cancelled) jobs, ended up re-running the whole thing >__<'

File: doc/userguide/rules/base64-keywords.rst [hunk 1]
```diff
@@ -10,6 +10,8 @@ base64_decode
 
 Decodes base64 data from a buffer and makes it available for the base64_data function.
 
+It is recommended that the transform ``from_base64`` be used instead of ``base64_decode``.
```
Comment (jufajardini) [inline doc/userguide/rules/base64-keywords.rst:13]: the `from_base64` here could be a `ref` to take the user directly to its section (for reference (heh): https://docs.ckan.org/en/2.9/contributing/documentation.html#cross-referencing-a-section-within-a-file)

**Review by jufajardini — CHANGES_REQUESTED** [review]
The doc contents look good, thinking that we could improve format, by adhering to our (attempted) guideline of having lines respect the 79/80 max character limit, where possible (meaning if readability would be compromised, like in the `Format` parts, I think we can leave as it is now) 

:)

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on SURI_TLPR1_alerts_cmp.
  

Pipeline 20416

File: doc/userguide/rules/base64-keywords.rst (hunk 1 above)
Comment (jlucovsky) [reply doc/userguide/rules/base64-keywords.rst:13]: Will do.

**Comment by jlucovsky** [comment]
Continued in #11007

**Pull request closed without merging**